src/Viewer.tsx

```diff
@@ -107,9 +108,18 @@ function Viewer(): ReactElement {
     return canvas;
   });
 
-  const [collection, setCollection] = useState<Collection | undefined>();
-  const [dataset, setDataset] = useState<Dataset | null>(null);
-  const [datasetKey, setDatasetKey] = useState("");
+  // TODO: Merge these into a single state slice call?
+  const collection = useViewerStateStore((state) => state.collection);
+  const setCollection = useViewerStateStore((state) => state.setCollection);
```

Comment:
why is the setter a separate state thing? is it an action and if so, do all state variables and actions get treated the same in this way?

Reply:
Yes, it's an action. All actions and state variables are pulled out of the store with this pattern currently!